test/test-network/systemd-networkd-tests.py

```diff
@@ -3027,8 +3025,20 @@ def test_dhcp_client_ipv4_only(self):
         self.assertRegex(output, r'192.168.5.7 proto dhcp scope link src 192.168.5.181 metric 1024')
         self.assertRegex(output, r'192.168.5.8 proto dhcp scope link src 192.168.5.181 metric 1024')
 
-    def test_dhcp_client_ipv4_use_routes_no(self):
-        copy_unit_to_networkd_unit_path('25-veth.netdev', 'dhcp-server-veth-peer.network', 'dhcp-client-ipv4-use-routes-no.network')
+    def test_dhcp_client_ipv4_use_routes_gateway(self):
+        for (routes, gateway) in itertools.product([True, False, None], repeat=2):
+            self.setUp()
+            with self.subTest(routes=routes, gateway=gateway):
+                self._test_dhcp_client_ipv4_use_routes_gateway(routes, gateway)
+            self.tearDown()
+
+    def _test_dhcp_client_ipv4_use_routes_gateway(self, routes, gateway):
+        testunits = ['25-veth.netdev', 'dhcp-server-veth-peer.network', 'dhcp-client-ipv4-use-routes-use-gateway.network']
+        if routes != None:
+            testunits.append('dhcp-client-ipv4-use-routes-use-gateway.network.d/use-routes-%s.conf' % ('yes' if routes else 'no'))
```

Comment:
Arrrgh, please don't add any %-formatting. (Esp. in this case: `'yes'` is a sequence, and the parentheses don't form a tuple, but pass through the sequence, which is then special-cased as a single argument. Blergh!)

Why not use a nice modern f-string?

Reply:
i *love* f-strings!  i don't always use them as sometimes i have to deal with older py3 or even py2, but I'm happy to add them in here :)